src/renderer/components/recording-page/recording-visualization.tsx

```diff
@@ -2,226 +2,161 @@ import log from 'electron-log';
 import React, { FC, useEffect, useRef } from 'react';
 import Col from 'react-bootstrap/Col';
 import Row from 'react-bootstrap/Row';
-import useInterval from 'react-use/lib/useInterval';
 import WaveSurfer from 'wavesurfer.js';
+import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone';
 import SpectrogramPlugin from 'wavesurfer.js/dist/plugin/wavesurfer.spectrogram';
 
 import { noOp } from '../../env-and-consts';
-import { checkFileExistence, join, readFile } from '../../utils';
+import { acquireAudioInputStream, checkFileExistence, join, readFile } from '../../utils';
+import { FileMonitor } from '../../utils/file-monitor';
+import { useAudioInputOutputDevices } from '../settings-page/hooks';
 
 import { State } from './types';
 
 interface RecordingVisualizationProps {
   prevState?: State;
   state: State;
   filePath?: string;
+  basePath: string;
+  recItemName: string;
 }
 
 const GRAPH_WIDTH = '60%';
-const GRAPH_HEIGHT = 80;
-const EMPTY_IMAGE_URL = 'data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw==';
 
-export const RecordingVisualization: FC<RecordingVisualizationProps> = ({ prevState, state, filePath }) => {
+export const RecordingVisualization: FC<RecordingVisualizationProps> = ({
+  prevState,
+  state,
+  filePath,
+  basePath,
+  recItemName,
+}) => {
   const waveSurferRef = useRef<WaveSurfer>();
   const waveformContainerRef = useRef<HTMLDivElement>(null);
-  const waveformImageRef = useRef<HTMLImageElement>(null);
   const spectrogramContainerRef = useRef<HTMLDivElement>(null);
-  const spectrogramImageRef = useRef<HTMLImageElement>(null);
-  // const rerender = useUpdate();
-
-  // useEffect(() => {
-  //   const [waveformContainer, waveformImage, spectrogramContainer, spectrogramImage] = [
-  //     waveformContainerRef.current,
-  //     waveformImageRef.current,
-  //     spectrogramContainerRef.current,
-  //     spectrogramImageRef.current,
-  //   ];
-  //
-  //   if (waveformContainer && waveformImage && spectrogramContainer && spectrogramImage) {
-  //     const waveformCanvases = waveformContainer.getElementsByTagName('canvas');
-  //     if (waveformCanvases.length) {
-  //       log.silly('Updating waveform image');
-  //       const waveform: HTMLCanvasElement = waveformCanvases[waveformCanvases.length - 1];
-  //       waveformImage.src = waveform.toDataURL();
-  //     }
-  //     const spectrogramCanvases = spectrogramContainer.getElementsByTagName('canvas');
-  //     if (spectrogramCanvases.length) {
-  //       log.silly('Updating spectrogram image');
-  //       const spectrogram: HTMLCanvasElement = spectrogramCanvases[spectrogramCanvases.length - 1];
-  //       spectrogramImage.src = spectrogram.toDataURL();
-  //     }
-  //   }
-  // });
+  const [, , audioInputDeviceId, , ,] = useAudioInputOutputDevices();
+  const fileMonitor = new FileMonitor(basePath);
+
+  function checkSpectrogramPlugin() {
```

Comment:
I think this is doing more than just "checking". Can you rename this to be more descriptive and/or add TS Doc?

Reply:
I added the TS Doc. I can't really think of a better name without the function name becoming really long. Do you have a name you were thinking of? Or is just the TS Doc enough?

Reply:
Add the doc first, then I can help think of a name.

Reply:
just pushed

Reply:
`checkSpectrogramPluginOrCreateNew`

Reply:
Also, prefer arrow functions:
```
const checkSpectrogramPluginOrCreateNew = () => {
```